report-service/src/main/java/gov/cdc/dataingestion/nbs/converters/Hl7ToRhapsodysXmlConverter.java

```diff
@@ -393,14 +413,38 @@ private HL7CNEType buildHL7CNEType(Cne cne) {
         HL7CNEType hl7CNEType = new HL7CNEType();
 
         hl7CNEType.setHL7Identifier(cne.getIdentifier());
-        hl7CNEType.setHL7Text(cne.getText());
-        hl7CNEType.setHL7NameofCodingSystem(cne.getNameOfAlternateCodingSystem());
-        hl7CNEType.setHL7AlternateIdentifier(cne.getAlternateIdentifier());
-        hl7CNEType.setHL7AlternateText(cne.getAlternateText());
-        hl7CNEType.setHL7NameofAlternateCodingSystem(cne.getNameOfAlternateCodingSystem());
-        hl7CNEType.setHL7CodingSystemVersionID(cne.getCodingSystemVersionId());
-        hl7CNEType.setHL7AlternateCodingSystemVersionID(cne.getAlternateCodingSystemVersionId());
-        hl7CNEType.setHL7OriginalText(cne.getOriginalText());
+
+        if ((null != cne.getText()) && (cne.getText().length() > 0)) {
```

Comment:
may be we can use common util function do null check

Reply:
Stefan suggested the same. Pls see my notes on his comments. Tx

Reply:
Done